calico_node/startup/startup.go

```diff
@@ -379,6 +380,18 @@ func autoDetectCIDR(method string, version int) *net.IPNet {
 		// Autodetect the IP by connecting a UDP socket to a supplied address.
 		destStr := strings.TrimPrefix(method, AUTODETECTION_METHOD_CAN_REACH)
 		return autoDetectCIDRByReach(destStr, version)
+	} else if strings.HasPrefix(method, AUTODETECTION_METHOD_SKIP_INTERFACE) {
+		// Autodetect the Ip by enumerating all interfaces (excluding
+		// known internal interfaces and any interfaces whose name
+		// matches the given regexes).
+		ifStr := strings.TrimPrefix(method, AUTODETECTION_METHOD_SKIP_INTERFACE)
+		// regexes are passed in as a string separated by ","
+		ifRegexes := strings.Split(ifStr, ",")
```

Comment:
Doesn't gain much, but you could do this as a single expression using regexp , something like:

`regexp.MustCompile("\s*,\s*").Split(ifStr, -1)`